tests/output.rs

```diff
@@ -1190,8 +1343,358 @@ fn test_byte_offset_only_matching() {
+            src/lib.rs:        assert_eq!(result, 4);
+            src/lib.rs:    }
+            src/stop.rs:fn stop_it() {}
+        "#,
+    );
+}
+
+#[test]
+fn test_column() {
+    assert_sorted_output(
+        "rust_project",
+        r#"
+            $ tree-sitter-grep -q '(function_item) @c' -l rust --column
+            src/stop.rs:1:1:fn stop_it() {}
+            src/helpers.rs:1:1:pub fn helper() {}
+            src/lib.rs:3:1:pub fn add(left: usize, right: usize) -> usize {
+            src/lib.rs:4:1:    left + right
+            src/lib.rs:5:1:}
+            src/lib.rs:12:5:    fn it_works() {
+            src/lib.rs:13:5:        let result = add(2, 2);
+            src/lib.rs:14:5:        assert_eq!(result, 4);
+            src/lib.rs:15:5:    }
+        "#,
+    );
+}
+
+#[test]
+fn test_column_context() {
+    assert_sorted_output(
+        "rust_project",
+        r#"
+            $ tree-sitter-grep -q '(function_item) @c' -l rust --column -A 1
+            src/stop.rs:1:1:fn stop_it() {}
+            --
+            src/helpers.rs:1:1:pub fn helper() {}
+            --
+            src/lib.rs:3:1:pub fn add(left: usize, right: usize) -> usize {
+            src/lib.rs:4:1:    left + right
+            src/lib.rs:5:1:}
+            src/lib.rs-6-
+            --
+            src/lib.rs:12:5:    fn it_works() {
+            src/lib.rs:13:5:        let result = add(2, 2);
+            src/lib.rs:14:5:        assert_eq!(result, 4);
+            src/lib.rs:15:5:    }
+            src/lib.rs-16-}
+        "#,
+    );
+}
+
+#[test]
+fn test_no_column() {
+    assert_sorted_output(
+        "rust_project",
+        r#"
+            $ tree-sitter-grep -q '(function_item) @c' -l rust --no-column
+            src/helpers.rs:pub fn helper() {}
+            src/lib.rs:pub fn add(left: usize, right: usize) -> usize {
+            src/lib.rs:    left + right
+            src/lib.rs:}
+            src/lib.rs:    fn it_works() {
+            src/lib.rs:        let result = add(2, 2);
+            src/lib.rs:        assert_eq!(result, 4);
+            src/lib.rs:    }
+            src/stop.rs:fn stop_it() {}
+        "#,
+    );
+}
+
+#[test]
+fn test_heading() {
+    assert_sorted_output(
+        "rust_project",
+        r#"
+            $ tree-sitter-grep -q '(function_item) @c' -l rust --heading
+            src/helpers.rs
+            pub fn helper() {}
+
+            src/stop.rs
+            fn stop_it() {}
+
+            src/lib.rs
+            pub fn add(left: usize, right: usize) -> usize {
+                left + right
+            }
+                fn it_works() {
+                    let result = add(2, 2);
+                    assert_eq!(result, 4);
+                }
+        "#,
+    );
+}
+
+#[test]
+fn test_pretty() {
+    assert_sorted_output(
+        "rust_project",
+        r#"
+             $ tree-sitter-grep -q '(function_item) @c' -l rust --pretty
+             [0m[1m[32msrc/stop.rs[0m
```

Comment:
You could picture having a "cuter" test-syntax for this

But not sure it's worth it?

Surprisingly this passed on Windows too I guess they use the same terminal color escape codes or whatever?

Reply:
Huh, I find that pretty surprising given how incompatible the terminals are in most other ways. (But they do somehow run git and whatever else, so maybe they are trying to be compatible.)

Reply:
> You could picture having a "cuter" test-syntax for this

I can't really 😄 — what were you thinking?

> But not sure it's worth it?

Yeah, doesn't seem like it's worth a ton of effort.